### PR TITLE
core(scripts): use `startLine`/`startColumn` to determine inline scripts

### PR DESCRIPTION
**Pull request by adamraine**
From https://github.com/GoogleChrome/lighthouse/pull/13793#discussion_r837962251

Instead of using the main doc url to find inline scripts, we can use the `startLine`/`startColumn`. This allows us to remove the dependency on the navigation url in treemap #13706.

### CODE REVIEW

File: lighthouse-core/audits/script-treemap-data.js [hunk 1]
```diff
@@ -166,26 +167,31 @@ class ScriptTreemapDataAudit extends Audit {
...
+
     for (const script of artifacts.Scripts) {
-      // Combine so that inline scripts show up as a single root node.
-      if (script.url === artifacts.URL.finalUrl) {
-        inlineScriptLength += (script.content || '').length;
-      }
+      if (!isInline(script)) continue;
+      let length = inlineNodeLengths.get(script.url) || 0;
+      length += script.length || 0;
+      inlineNodeLengths.set(script.url, length);
```
Comment (connorjclark) [inline lighthouse-core/audits/script-treemap-data.js:179]: Can you undo these changes? I am reworking inline scripts here: https://github.com/GoogleChrome/lighthouse/pull/13802/

Comment (adamraine) [reply lighthouse-core/audits/script-treemap-data.js:179]: Honestly the treemap stuff is why this is a separate PR from #13793. I'll just close and move the unminified js stuff over there.
